gax/src/test/java/com/google/api/gax/batching/v2/BatcherImplTest.java

```diff
@@ -268,4 +445,53 @@ private void testElementTriggers(BatchingSettings settings) throws Exception {
     assertThat(result.get()).isEqualTo(16);
     assertThat(anotherResult.isDone()).isTrue();
   }
+
+  private class ElementAddThread extends Thread {
+    private volatile boolean isCancelled;
+    private final AtomicInteger counter = new AtomicInteger();
```

Comment:
Why does this need to atomic? Can’t it just be a local int? Also I would just inline this

Reply:
I am sorry again. 
Initially, I was using this counter to step check(say per 100 elements) in the test case. I removed this now.